### ISSUE TITLE
Handle editing "partOf"-type sources

### ISSUE THREAD

**Issue by ptgolden** [periodo/periodo-client#29]

**Comment by ptgolden** [comment]
To be clear, this is only an issue for non-linked data sources. I actually don't know if we want to go down the rabbit hole of precisely describing sources. Part-of relationships for non-linked data sources should probably be included in the `citation` field.

Does that sound sensible, @rybesh?

**Comment by rybesh** [comment]
I'm not clear on what the issue is here. What `citation` field?

**Comment by atomrab** [comment]
Is this about subperiods in a single publication, like Shelmerdine has Late Helladic, within which is Late Helladic III, within which is Late Helladic IIIA?

**Comment by rybesh** [comment]
No, I think it is related to the way we model the locations of period definitions (e.g. page numbers) within a source.

**Comment by atomrab** [comment]
Ah hah. I've got nothin', then.

**Comment by ptgolden** [comment]
> No, I think it is related to the way we model the locations of period definitions (e.g. page numbers) within a source.

I'm referring to when sources themselves are made up of a locator and a "partOf" relationship

**Comment by ptgolden** [comment]
For example:

```
{
      ...

      "id": "https://perio.do/.well-known/genid/assigned/p0244q7",
      "source": {
        "locator": "page 561",
        "partOf": {
          "contributors": [
            {
              "id": "http://viaf.org/viaf/66565783",
              "name": "Hornblower, Simon."
            },
            {
              "id": "http://viaf.org/viaf/29582600",
              "name": "Spawforth, Antony (Antony J.S.)"
            }
          ],
          "id": "http://www.worldcat.org/oclc/783109160",
          "title": "The Oxford classical dictionary.",
          "yearPublished": 2003
        }
      },
      "type": "PeriodCollection"
    }
````

**Comment by ptgolden** [comment]
Or, this source, with nested partOf relationships:

```
{
      ...
      "id": "https://perio.do/.well-known/genid/assigned/p0ms2ch",
      "source": {
        "locator": "page 319",
        "partOf": {
          "creators": [
            {
              "name": "W. A. McDonald"
            },
            {
              "name": "W. D. E. Coulson"
            }
          ],
          "partOf": {
            "contributors": [
              {
                "id": "http://viaf.org/viaf/111563392",
                "name": "MacDonald, William A."
              },
              {
                "id": "http://viaf.org/viaf/46784178",
                "name": "Coulson, William D.E."
              }
            ],
            "id": "http://www.worldcat.org/oclc/58641827",
            "title": "Excavations at Nichoria in Southwest Greece. 3, Dark age and Byzantine occupation.",
            "yearPublished": 1983
          },
          "title": "The Dark Age at Nichoria: A Perspective"
        }
      },
```

**Comment by ptgolden** [comment]
I created this issue because the "partOf" relationships in sources (not in period definitions) seem like they are trying to express part-whole relationships between subsections of source material. I was considering whether we should expose expressing such part-whole relationships for sources that do not have linked data representations.

The `citation` field is used in non-linked data sources to create a full-text citation that describes the source beyond the structured inputs provided (title, contributors, year published). We at one point decided to provide that kind of free input instead of trying to completely model the kinds of sources people might want to enter into the client interface. My suggestion in my second comment was that "partOf" relationships for non-linked data sources should be entered into that field rather than explicitly supported.

**Comment by atomrab** [comment]
Oh, I totally get it now. This is about book chapters/contributions to
edited volumes, where the edited book has a WorldCat identifier, but the
section doesn't, since it's neither an article nor a book itself.

These sections will never have their own URIs. The primary thing for me is
to preserve a) the URI for the book itself (to keep the LD thing in the
forefront) and b) the authorship of the section where the period assertion
appears, since it's the author(s) of the section who are the authorities
for the assertion, not the editors. This is sort of half-way between linked
data (the book with its OCLC number) and non-linked data (the
separately-authored chapter).

As long as both of these elements appear, I'm ok with whatever the two of
you come up with on a data-model level. But I think that both of these
elements are fundamental, and a solution that makes it impossible to
include both is a problem. Free text is fine for this, though again the
book itself should get its WorldCat link.

**Comment by ptgolden** [comment]
I won't be able to implement such a thing in the interface now, so I'm going to leave these situations to a free text field. If someone wants to add any sort of linked data source beyond what the client accepts, they can create and submit a JSONPatch outside of the client.

In future development, it would be good to be able to mix LD and non-LD sources in such a way.

**Comment by atomrab** [comment]
Can we have the regular LD biblio lookup, and then a notes field in which
the free text of the chapter could be entered?

**Comment by ptgolden** [comment]
Yes, a "notes" field for sources already exists

**Comment by rybesh** [comment]
This issue is very unclear to me; I can't tell what is being proposed.

**Comment by atomrab** [comment]
The problem is providing structured bibliographic data for chapters in edited volumes. The chapters don't have worldcat URIs, so if you just do the chapter, you lose the Linked Data component. But if you just provide the book URI, you lose the actual chapter source. 

We can put the chapter in the general "editorial notes" field. But this doesn't strictly correspond to our concept of "editorial notes", since we'd actually just be expanding the bibliographic citation, not adding commentary. I suppose this is fine, too -- except that it won't be easy to pull out the authors as LD. 

Would it make sense to have an optional "additional contributors" field where chapter authors could be listed, and at some later date linked (by us or by others) to their VIAF ids or other URIs? My concern here is that intellectual genealogy will be harder to reconstruct if we are only looking at editors, and it will be hard to parse additional contributors from unstructured editorial notes. Does that make more sense?

**Comment by rybesh** [comment]
> The problem is providing structured bibliographic data for chapters in edited volumes.

I think this is out of scope for PeriodO—there are other projects working on this general problem, and we should adopt their solutions when they are ready. In the meantime we can put further information where needed as unstructured data. As always our goal should be to provide useful pointers to finding a copy of the source, not to be an authoritative source of structured bibliographic data.
